src/filebrowser_upload/main.py

```diff
@@ -87,31 +184,108 @@ def __exit__(self, *exc):
         self.close()
 
 
-def upload(CONFIG):
-    url = get_upload_url(CONFIG)
-    print('Upload to {}'.format(url))
-    try:
-        token = get_token(CONFIG)
-    except requests.exceptions.HTTPError as ex:
-        print('Login failed: {} {}'.format(ex.response.status_code, ex.response.reason))
-        return
-    override = 'true' if CONFIG.override else 'false'
-    fileobj = open(CONFIG.filepath, 'rb')
-    if not CONFIG.no_progress:
+def upload(file, url, no_progress, override, headers, insecure, report):
+    """
+    Upload file to the specified url.
+
+    Args:
+        file (str): File to be read.
+        url (str): File destination url.
+        no_progress (bool): Enable or disable progress bar.
+        override (bool): Override file or not.
+        headers (dict): Authentication headers.
+        insecure (bool): Allow insecure server connections when using SSL.
+        report (dict): Update upload report.
+    """
+
+    fileobj = open(file, "rb")
+
+    if not no_progress:
         fileobj = ProgressFile(fileobj)
-    headers = {
-        'X-Auth': token,  # version >= 2.0.3 seems use this header
-        'Authorization': 'Bearer {}'.format(token),  # version <= 2.0.0 seems use this header
-    }
+
     with fileobj:
         response = requests.post(
-            url, data=fileobj,
+            url,
+            data=fileobj,
             params={"override": override},
             headers=headers,
-            verify=not CONFIG.insecure,
+            verify=not insecure,
         )
-    print('{} {}'.format(response.status_code, response.reason))
+
+    report_key = f"{response.status_code} {response.reason}"
+    print(f"{report_key}\n")
+
+    if report.get(report_key):
```

Comment:
how about exit (raise Exception) when failed?

Reply:
This report was introduced to just give to the user a brief summary of what happened. 
Raise an exception and exit in case of a "big" folder upload is not a proper way imho.

But this report is just a basic idea that could evolve further.